rewriter/flatten.cc

```diff
@@ -351,8 +378,15 @@ class FlattenWalk {
         auto &methods = curMethodSet();
         // if we get a MethodData back, then we need to move this and replace it
         if (auto md = methods.popScope()) {
-            // we'll replace MethodDefs with the symbol that corresponds to the name of the method
-            auto replacement = ast::MK::Symbol(methodDef->loc, methodDef->name);
+            unique_ptr<ast::Expression> replacement;
+
+            if (methods.atTopLevel()) {
+                replacement = ast::MK::EmptyTree();
+            } else {
+                // we'll replace MethodDefs with the symbol that corresponds to the name of the method
+                replacement = ast::MK::Symbol(methodDef->loc, methodDef->name);
+            }
```

Comment:
This will only rewrite a method to a symbol if it's in a position where a symbol might be actually used (e.g. the `MethodDef` is an argument to a send.)

Reply:
Seems like ad-hoc optimization to me, but sure.
CFG finalization passes before typechecking have this optimization as a proper pass.